apollo-gradle-plugin-kotlin/src/main/kotlin/com/apollographql/apollo/gradle/kotlin/ApolloExtension.kt

```diff
@@ -0,0 +1,117 @@
+package com.apollographql.apollo.gradle.kotlin
+
+import com.apollographql.apollo.compiler.NullableValueType
+import org.gradle.api.Action
+
+open class ApolloExtension {
```

Comment:
Question, current implementation of plugin invalidates `UP-TO-DATE` state if any changes made to the plugin configuration (don't remember how the tracking exactly works in Gradle), does this plugin provides the same feature? Like if we change the configuration for custom scalar type mapping will Gradle detect and invalidate the cache and re-run code gen?

Reply:
I'll add a test for that.

Reply:
The test is already there actually :) https://github.com/martinbonnin/apollo-android/blob/a651a10cd6ddf00a160defcc6d97f6b08e063243/apollo-gradle-plugin-kotlin/src/test/kotlin/com/apollographql/apollo/gradle/AndroidLibraryTests.kt#L87

I guess this works because `customTypeMapping` is marked as @Input to the task ? So if it changes the task will be re-run ? Gradle magic :sparkles: